### DISCUSSION TITLE
Separate OTLP proto from the transport like GRPC or HTTP?

### DISCUSSION THREAD

**Discussion by ThomsonTan** [open-telemetry/opentelemetry-cpp-contrib#410] [Ideas]
The OTLP proto library is mostly used by the OTLP gRPC or HTTP exporter, but the proto library can also be used with other protocols, like in the [user_events exporter](https://github.com/open-telemetry/opentelemetry-cpp-contrib/blob/main/exporters/user_events/src/metrics_exporter.cc#L68).

Could we add a flash like `WITH_OTLP_PROTO` which builds the proto build, but don't enable the gRPC or HTTP exporter?

The current workaround used in the above user_events exporter is it enables OTLP/HTTP exporter as implicit dependence, even it doesn't use it really.

**Comment by lalitb** [comment]
Is this question for adding the cmake option in otel-cpp build? Good to ask there in that case ;)

**Reply by ThomsonTan** [reply]
Yes, copied the ask to the otel-cpp repo.
